### PR TITLE
UNDERC depth data and crosswalk, Isle Royale hypso data and crosswalk

### PR DESCRIPTION
**Pull request by jread-usgs**
I put both issue #261 and #176 in here. #261 is only partially resolved because the temperature data are not yet in the pipeline and the light attenuation data (additional processing of "PAR" profiles in the same dataset as the temperature) are required to gain more lakes. 

Despite the UNDERC dataset having 100+ lakes, we only gained one from that depth dataset. The others were either already in the pipeline or not in but still missing Kw/secchi data so not considered a "GLM" lake at this point, which is our lowest threshold for modeling. 

The Isle Royale (ISRO) dataset added more, since the pipeline seemingly already had temperature and secchi data for these sites (likely from WQP) and so adding hypsography resulted in ten new GLM lakes and eight new PGDL lakes. Also, since these lakes seem to be highly observed, it is worth noting that I only added the low hanging fruit contour lakes which could be easily turned into hypso. I didn't try to handle multilines so I'm not sure if issues like #167 will exist or if these could be cast into polygons without issue. We probably want to spin off another lower priority issue for the rest of the ISRO lakes. 

image

### CODE REVIEW

**Comment by jread-usgs** [comment]
Oh yeah, a zoom in on the island 
image

for #176

File: 1_crosswalk_fetch.yml [hunk 1]
```diff
@@ -127,6 +128,16 @@ targets:
     command: slice_iadnr_contour(target_name,
       contour_ind = '1_crosswalk_fetch/out/iadnr_contours_sf.rds.ind')
 
+  1_crosswalk_fetch/out/isro_contours_sf.rds.ind:
+    command: fetch_isro_lakes(target_name,
+      use_geoms = I('Multi Polygon'),
+      gdb_filename = I('ISRO_GLKN_BathymetricData.gdb'),
+      zip_ind = '3_params_fetch/in/ISRO_GLKN_Bathymetric_FinalData.zip.ind')
+
+  1_crosswalk_fetch/out/isro_lakes_sf.rds.ind:
```
Comment (lindsayplatt) [inline 1_crosswalk_fetch.yml:137]: Should this get added under the overall makefile depends list for `1_crosswalk_fetch`?

File: 1_crosswalk_fetch/src/fetch_crosswalk.R [hunk 2]
```diff
@@ -32,6 +32,54 @@ process_wbic_lakes <- function(file_out = '1_crosswalk_fetch/out/wbic_lakes_sf.r
   saveRDS(wbic_lakes, file = file_out)
 }
 
+fetch_isro_lakes <- function(out_ind, zip_ind, remove_IDs = c(), gdb_filename, use_geoms){
```
Comment (lindsayplatt) [inline 1_crosswalk_fetch/src/fetch_crosswalk.R:35]: Note that `remove_IDs` is not used in the function code at all

File: 1_crosswalk_fetch/src/fetch_crosswalk.R [hunk 3]
```diff
@@ -32,6 +32,54 @@ process_wbic_lakes <- function(file_out = '1_crosswalk_fetch/out/wbic_lakes_sf.r
...
+    dplyr::select(layer_name, depth_col)
+
+  data_clean <- purrr::pmap(use_layers, function(layer_name, depth_col){
+    # get "Beaver" from "Beaver_CntrPolys" etc:
+    lake_name <- strsplit(layer_name, '_')[[1]][1]
+
+    st_read(file.path(gdb_path, gdb_filename), layer_name, quiet = TRUE) %>%
+      sf::st_zm() %>% st_transform(crs = 4326) %>%
+      rename(geometry = Shape) %>%
+      mutate(site_id = sprintf('isro_%s', lake_name)) %>%
```
Comment (lindsayplatt) [inline 1_crosswalk_fetch/src/fetch_crosswalk.R:70]: This is lowercase but later in `2_crosswalk_munge` you pass in `ISRO_` to `poly1_ID_name`. Does that matter?

File: 7a_temp_coop_munge/log/7a_temp_coop_munge_tasks.ind [hunk 4]
```diff
@@ -1,153 +1,154 @@
-7a_temp_coop_munge/tmp/1945_2020_All_MNDNR_MPCA_Temp_DO_Profiles.rds.ind: ab2f36cec6a8803f1aa28f0ebe4d1d47
```
Comment (lindsayplatt) [inline 7a_temp_coop_munge/log/7a_temp_coop_munge_tasks.ind:1]: I wonder why all of these hashes changed?

**Review by lindsayplatt — APPROVED** [review]
Ran `scmake('8_viz/out/lakes_summary_fig.html.ind')` and it did what I expected up until it started to build `7_drivers_munge/out/7_drivers_munge_tasks.ind`, so I stopped it.

File: 1_crosswalk_fetch/src/fetch_crosswalk.R (hunk 3 above)
Comment (jread-usgs) [reply 1_crosswalk_fetch/src/fetch_crosswalk.R:70]: I don't think we're consistent on this in the repo...but the reason for this is that `isro_{asdfasdf}` is the ISRO_ID field value, while the `ISRO_ID` that appears as an input value for `poly1_ID_name` in the `crosswalk_poly_intersect_poly()` function is the name of the field/column. I think for the most part the field/column name for these other IDs are all in caps, and for the most part (but I'm sure there are exceptions) the _value_ in that column is prefixed with the lowercase of it.

Comment (lindsayplatt) [reply 1_crosswalk_fetch/src/fetch_crosswalk.R:70]: Hmmm ok, I think [the value for the UNDERC is capitalized](https://github.com/USGS-R/lake-temperature-model-prep/pull/327/files#diff-4ae5528d8c4172ac5cc98d0b5259d450cf4be0ba079243f68e54dc33a295c859R311). Sounds like this isn't a hidden bug or anything, just more of a style issue.

**Pull request merged**